lib/chat/ChatPage.dart

```diff
@@ -1,39 +1,171 @@
+  static const dismissModalIcon = "assets/raw/nav_icon_cancel.png";
+  static const optionButtonIcon = "assets/icons/nav_icon_options.png";
+}
+
+class _LocalisedStrings {
+  static resetProcess() => Intl.message("Reset Process");
+
+  static skip() => Intl.message("Skip");
+
+  static cancel() => Intl.message("Cancel");
+
+  static confirm() => Intl.message("Confirm Details");
+}
+
+class ChatPage extends StatefulWidget {
+  @override
+  _ChatPageState createState() => _ChatPageState();
+}
+
+class _ChatPageState extends State<ChatPage> {
+  @override
+  void initState() {
+    super.initState();
+  }
+
   @override
   Widget build(BuildContext context) {
-    return ChatFlowFromFile(context).start(
-      file: _createChatFile(),
-      onSuccess: (map) {
-        _doOnSuccess(context, map);
-      },
-      onError: (error) {
-        _doOnError(context, error);
-      },
+    return Scaffold(
+      appBar: _buildSimpleAppBar(context),
+      body: _buildChat(context),
     );
   }
 
-  File _createChatFile() {
-    return File(_Constants.defaultPathToJSONFile);
+  AppBar _buildSimpleAppBar(BuildContext context) {
+    return AppBar(
+      elevation: _Constants.appBarElevation,
+      leading: _buildCloseButton(context),
+      actions: <Widget>[_buildOptionsButton(context)],
+    );
   }
 
-  _doOnSuccess(BuildContext context, Map<String, String> map) {
+  _buildChat(BuildContext context) => ChatFlowFromFile(context).start(
+        file: _createChatFile(),
+        onSuccess: (map) {
+          _doOnSuccess(context, map);
+        },
+        onError: (error) {
+          _doOnError(context, error);
+        },
+      );
+
+  _buildCloseButton(BuildContext context) => FlatButton(
+        onPressed: () => _navigateBack(context),
+        padding: _Constants.appBarEdgePadding,
+        child: Image.asset(
+          _Assets.dismissModalIcon,
+          height: _Constants.appBarIconSize,
+          width: _Constants.appBarIconSize,
+        ),
+      );
+
+  _buildOptionsButton(BuildContext context) => Center(
+        child: FlatButton(
+          onPressed: () => _optionsTapped(
+            _optionsMenu(context),
+            context,
+          ),
+          child: Image.asset(
+            _Assets.optionButtonIcon,
+            width: _Constants.appBarIconSize,
+            height: _Constants.appBarIconSize,
+          ),
+        ),
+      );
+
+  File _createChatFile() => File(_Assets.defaultPathToJSONFile);
+
+  _doOnSuccess(BuildContext context, Map<String, ChatResponseViewModel> map) {
     print("On Success recevied: ${map.toString()}");
-    Navigator.of(context).pop();
+
+    NavigationScope.presentModal(
+      context,
+      FarmDetails(
+        viewModel: FarmDetailsViewModel(
+          items: _getFarmDetailsListFromMap(map),
+          buttonTitle: _LocalisedStrings.confirm(),
+          confirm: () {
+            _navigateBack(context);
+            _navigateBack(context);
```

Comment:
We need to call twice at `navigateBack` ?

Reply:
Well at this moment yes we should, the reason is that we have two opened modals and we want on confirmation press just dismiss the firsts and the second one.
Not sure if in flutter we can pop back to some specific stacked instance instead of doing double pop back...

Reply:
must be a better way

Reply:
Apparently we can do so, but using **pushNamed** which allows to push new screens with given name. On pop you can specify to which route you want to pop back.
But we have two problems here:

- In code base we are not using **pushNamed**
- Even if we add this logic, for chat we are using the modal which is alert dialog popup and it handles the navigation by itself, so we have no pop back name to be referenced in here.